drivers/SmartThings/zigbee-thermostat/src/aqara/init.lua

```diff
@@ -0,0 +1,290 @@
+-- Copyright 2024 SmartThings
+--
+-- Licensed under the Apache License, Version 2.0 (the "License");
+-- you may not use this file except in compliance with the License.
+-- You may obtain a copy of the License at
+--
+--     http://www.apache.org/licenses/LICENSE-2.0
+--
+-- Unless required by applicable law or agreed to in writing, software
+-- distributed under the License is distributed on an "AS IS" BASIS,
+-- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+-- See the License for the specific language governing permissions and
+-- limitations under the License.
+local data_types = require "st.zigbee.data_types"
+local clusters = require "st.zigbee.zcl.clusters"
+local cluster_base = require "st.zigbee.cluster_base"
+local capabilities = require "st.capabilities"
+local Thermostat = clusters.Thermostat
+local PowerConfiguration = clusters.PowerConfiguration
+
+local invisibleCapabilities = capabilities["stse.invisibleCapabilities"]
+local valveCalibration = capabilities["stse.valveCalibration"]
+local valveCalibrationCommandName = "startCalibration"
+
+local PRIVATE_CLUSTER_ID = 0xFCC0
+local MFG_CODE = 0x115F
+local PRIVATE_VALVE_CALIBRATION_ID = 0x0270
+local PRIVATE_VALVE_SWITCH_ATTRIBUTE_ID = 0x0271
+local PRIVATE_THERMOSTAT_OPERATING_MODE_ATTRIBUTE_ID = 0x0272
+local PRIVATE_THERM0STAT_VALVE_DETECTION_SWITCH_ID = 0x0274
+local PRIVATE_THERMOSTAT_ALARM_INFORMATION_ID = 0x0275
+local PRIVATE_CHILD_LOCK_ID = 0x0277
+local PRIVATE_ANTIFREEZE_MODE_TEMPERATURE_SETTING_ID = 0x0279
+local PRIVATE_VALVE_RESULT_CALIBRATION_ID = 0x027B
+local PRIVATE_BATTERY_ENERGY_ID = 0x040A
+
+local FINGERPRINTS = {
+    { mfr = "LUMI", model = "lumi.airrtc.agl001" }
+}
+
+local preference_map = {
+  ["stse.notificationOfValveTest"] = {
+    cluster_id = PRIVATE_CLUSTER_ID,
+    attribute_id = PRIVATE_THERM0STAT_VALVE_DETECTION_SWITCH_ID,
+    mfg_code = MFG_CODE,
+    data_type = data_types.Uint8,
+    value_map = { [true] = 0x01,[false] = 0x00 },
+  },
+  ["stse.antifreezeModeSetting"] = {
+    cluster_id = PRIVATE_CLUSTER_ID,
+    attribute_id = PRIVATE_ANTIFREEZE_MODE_TEMPERATURE_SETTING_ID,
+    mfg_code = MFG_CODE,
+    data_type = data_types.Uint32,
+    value_map = {
+      ["1"] = 500,
+      ["2"] = 550,
+      ["3"] = 600,
+      ["4"] = 650,
+      ["5"] = 700,
+      ["6"] = 750,
+      ["7"] = 800,
+      ["8"] = 850,
+      ["9"] = 900,
+      ["10"] = 950,
+      ["11"] = 1000,
+      ["12"] = 1050,
+      ["13"] = 1100,
+      ["14"] = 1150,
+      ["15"] = 1200,
+      ["16"] = 1250,
+      ["17"] = 1300,
+      ["18"] = 1350,
+      ["19"] = 1400,
+      ["20"] = 1450,
+      ["21"] = 1500
+    },
```

Comment:
you could make `value_map` a function instead that returns `tonumber(value)*50 + 450` and then in the info changed handler, check the type of the `value_map` right after you check for `nil`.

Reply:
Thx，pls check again~

Reply:
Thx，pls check again~